benchmarks/src/main/scala/zio/GenBenchmarks.scala

```diff
@@ -0,0 +1,81 @@
+package zio
+
+import java.util.concurrent.TimeUnit
+
+import hedgehog.core.GenT
+import org.openjdk.jmh.annotations._
+import org.scalacheck
+import zio.random.Random
+import zio.test.Gen
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+class GenBenchmarks {
+
+  @Param(Array("1000"))
+  var size: Int = _
+
+  @Benchmark
+  def zioDouble: Gen[Random, Double] =
+    Gen.double(0, size)
+
+  @Benchmark
+  def zioIntListOfSizeN: Gen[Random, List[Int]] =
+    Gen.listOfN(size)(Gen.anyInt)
+
+  @Benchmark
+  def zioStringOfSizeN: Gen[Random, String] =
+    Gen.stringN(size)(Gen.anyChar)
+
+  @Benchmark
+  def scalaRandomDouble: Double =
+    scala.util.Random.nextDouble()
+
+  @Benchmark
+  def scalaRandomIntListOfSizeN: List[Int] =
+    List.fill(size)(scala.util.Random.nextInt())
+
+  @Benchmark
+  def scalaRandomStringOfSizeN: String =
+    scala.util.Random.alphanumeric.take(size).mkString
+
+  @Benchmark
+  def scalaCheckDouble: scalacheck.Gen[Double] =
+    scalacheck.Gen.choose(0, size)
+
+  @Benchmark
+  def scalaCheckIntListOfSizeN: scalacheck.Gen[List[Int]] =
+    scalacheck.Gen.listOfN(size, scalacheck.Gen.choose(0, size))
+
+  @Benchmark
+  def scalaCheckStringOfSizeN: scalacheck.Gen[String] =
+    scalacheck.Gen.listOfN(size, scalacheck.Gen.alphaChar).map(_.mkString)
```

Comment:
These are all just benchmarking the cost of creating the `Gen`, not the cost of creating N number of values using the `Gen`. I suggest to change to the latter since that's what matters for performance of the tests.